src/chessboard/components/Piece.tsx

```diff
@@ -118,13 +118,15 @@ export function Piece({
       newSquare &&
       !isPremovedPiece
     ) {
-      const { sourceSq, targetSq } = getSquareCoordinates(square, newSquare[0]);
+      const sourceSq = square;
+      const targetSq = newSquare[0];
       if (sourceSq && targetSq) {
+        const squareWidth = boardWidth / 8;
         setPieceStyle((oldPieceStyle) => ({
           ...oldPieceStyle,
-          transform: `translate(${targetSq.x - sourceSq.x}px, ${
-            targetSq.y - sourceSq.y
-          }px)`,
+          transform: `translate(${
+            (targetSq.charCodeAt(0) - sourceSq.charCodeAt(0)) * squareWidth
+          }px, ${(Number(sourceSq[1]) - Number(targetSq[1])) * squareWidth}px)`,
```

Comment:
why did this need to be changed?

Reply:
`getBoundingClientRect()` does not calculate properly the {x,y} coordinates when we add  `transform: rotateX(27deg)`. The animation of piece movements becomes not correct

Reply:
See how black pieceses move without these changes


https://github.com/Clariity/react-chessboard/assets/42891870/8437df2f-0aae-4a4a-a278-760b6e3d7fff